shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -292,46 +292,27 @@ - (void)dealloc {
                                                       object:self
                                                     userInfo:nil];
 
-  // It will be destroyed and invalidate its weak pointers
-  // before any other members are destroyed.
-  _weakFactory.reset();
-
   /// nil out weak references.
```

Comment:
TODO + tracking issue for eliminating this code by ensuring that `FlutterEngineRegistrar` is using `weak` pointers?